react/react2/inclass/app/form2/page.jsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+
+import { useRef } from "react";
+import styles from "./page.css";
+export default function SubmitNote() {
+  const refs = {
+    firstName: useRef("firstName"),
+    lastName: useRef("lastName"),
+    email: useRef("email"),
+    phoneNumber: useRef("phoneNumber"),
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const formData = {};
+
+    for (let key in refs) {
+      formData[key] = refs[key].current.value;
+      console.log(formData);
+    }
+  };
+
+  const validateEmail = () => {
+    const emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
+    const email = emailRef.current.value;
+    setIsValid(emailRegex.test(email));
```

Comment:
Idea is good, but it doesn't work. Also i don't see definition `setIsValid` and using this validation data for highlighting an error 

<img width="490" alt="image" src="https://github.com/user-attachments/assets/6ac672c4-eba9-4f00-9975-a76521d3dc3d">